### PR TITLE
[MediaVision] Add APIs for Design QR feature

### PR DESCRIPTION
**Pull request by hsgwon**
### Description of Change ###

Add APIs for Design QR feature

### API Changes ###

 - TCSACR-534

### CODE REVIEW

**Review by aferin — APPROVED** [review]

File: src/Tizen.Multimedia.Vision/MediaVision/QrConfiguration.cs [hunk 1]
```diff
@@ -78,5 +82,67 @@ public QrConfiguration(QrMode qrMode, ErrorCorrectionLevel ecc, int version)
...
+                _dataShape = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the finder shape of the Design QR code.
+        /// </summary>
+        /// <remarks>The default value is <see cref="QrShape.Rectangular"/>.</remarks>
+        /// <remarks>If user get value before setting value, FinderShape will return null.</remarks>
+        /// <exception cref="ArgumentException"><paramref name="value"/> is not valid.</exception>
```
Comment (wiertel) [inline src/Tizen.Multimedia.Vision/MediaVision/QrConfiguration.cs:132]: maybe
```suggestion
        /// <exception cref="ArgumentException"><paramref name="value"/> set is not valid.</exception>
```
if you decide to apply that, then the same for lines 89, 90, 111

File: src/Tizen.Multimedia.Vision/MediaVision/QrShape.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    /// <since_tizen> 11 </since_tizen>
+    public enum QrShape
+    {
+        /// <summary>
+        /// Rectangular.
+        /// </summary>
+        Rectangular,
+
+        /// <summary>
+        /// Round rectangular.
```
Comment (wiertel) [inline src/Tizen.Multimedia.Vision/MediaVision/QrShape.cs:33]: maybe "Rounded rectangular", "Rectangular with rounded corners"

File: src/Tizen.Multimedia.Vision/MediaVision/QrConfiguration.cs [hunk 3]
```diff
@@ -78,5 +82,67 @@ public QrConfiguration(QrMode qrMode, ErrorCorrectionLevel ecc, int version)
...
+            set
+            {
+                ValidationUtil.ValidateEnum(typeof(QrShape), value, nameof(value));
+
+                _dataShape = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the finder shape of the Design QR code.
```
Comment (wiertel) [inline src/Tizen.Multimedia.Vision/MediaVision/QrConfiguration.cs:128]: the QrConfiguration is described as "Represents a QR configuration of Barcode Generator", does this option belong to generator?

File: src/Tizen.Multimedia.Vision/MediaVision/QrConfiguration.cs [hunk 4]
```diff
@@ -78,5 +82,67 @@ public QrConfiguration(QrMode qrMode, ErrorCorrectionLevel ecc, int version)
         /// </summary>
         /// <since_tizen> 4 </since_tizen>
         public int Version { get; }
+
+        /// <summary>
+        /// Gets or sets the embed image absolute path of the Design QR code.
```
Comment (wiertel) [inline src/Tizen.Multimedia.Vision/MediaVision/QrConfiguration.cs:87]: is "Design QR" some special term?  "embed" is a verb so I'm not sure this is correct. Please describe to me how does this work and I will present some suggestions for rewording. While we are at it, you could explain that Data and Finder shapes are.

**Review by wiertel — CHANGES_REQUESTED** [review]

File: src/Tizen.Multimedia.Vision/MediaVision/QrConfiguration.cs [hunk 5]
```diff
@@ -78,5 +82,67 @@ public QrConfiguration(QrMode qrMode, ErrorCorrectionLevel ecc, int version)
...
+
+                _embedImagePath = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the data shape of the Design QR code.
+        /// </summary>
+        /// <remarks>The default value is <see cref="QrShape.Rectangular"/>.</remarks>
+        /// <remarks>If user get value before setting value, DataShape will return null.</remarks>
```
[the remark this answers is not in the document]
Comment (hsgwon) [reply src/Tizen.Multimedia.Vision/MediaVision/QrConfiguration.cs:110]: I wrote the second line when `_dataShape` was nullable type. I'll remove it.

File: src/Tizen.Multimedia.Vision/MediaVision/QrShape.cs (hunk 2 above)
Comment (hsgwon) [reply src/Tizen.Multimedia.Vision/MediaVision/QrShape.cs:33]: I'll change it.

File: src/Tizen.Multimedia.Vision/MediaVision/QrConfiguration.cs (hunk 3 above)
Comment (hsgwon) [reply src/Tizen.Multimedia.Vision/MediaVision/QrConfiguration.cs:128]: Right.

File: src/Tizen.Multimedia.Vision/MediaVision/QrConfiguration.cs (hunk 4 above)
Comment (KwangSon) [reply src/Tizen.Multimedia.Vision/MediaVision/QrConfiguration.cs:87]: > is "Design QR" some special term? 

Yes it is. It just 
> "embed" is a verb so I'm not sure this is correct. Please describe to me how does this work and I will present some suggestions for rewording. While we are at it, you could explain that Data and Finder shapes are.

EmbedImagePath is kinds of the logo image that goes in the middle of the QR code.
you could see more deatil in here
https://github.com/Samsung/tizen-docs/blob/tizen_7.5_prepare/docs/application/native/guides/multimedia/image-barcode.md#design-qr

File: src/Tizen.Multimedia.Vision/MediaVision/QrConfiguration.cs (hunk 1 above)
Comment (hsgwon) [reply src/Tizen.Multimedia.Vision/MediaVision/QrConfiguration.cs:130]: I don't understand the meaning of `set` in here.

**Review by wiertel — APPROVED** [review]

**Pull request merged**